### DISCUSSION TITLE
CLI command to output formatting class order

### DISCUSSION THREAD

**Discussion by btkostner** [tailwindlabs/tailwindcss#7242] [Ideas]
First off I just want to say that https://github.com/tailwindlabs/prettier-plugin-tailwindcss is amazing, I love it, and it fixes one of my biggest pain points! 🎊 

Some languages have their own way to format files. Elixir for example has [`mix format`](https://hexdocs.pm/mix/main/Mix.Tasks.Format.html) which allows [writing plugins](https://hexdocs.pm/mix/main/Mix.Tasks.Format.html#module-plugins) to extend it. I'd love to write one for Tailwind that duplicates most of the `prettier-plugin-tailwindcss` package. Sadly determining the order of classes requires using some JS, which is a problem for languages that use the standalone CLI.

I propose adding a cli command that outputs all of the classes in order. That way third party plugins can integrate into tailwind formatting (node or standalone.)

**Comment by btkostner** [comment]
Adding to this, there are now a couple of third party sort programs for various languages. Most of them seem to define a hard coded list of classes for sorting. I'd be nice to get a handle on this before it gets too out of hand.

Rust: https://github.com/avencera/rustywind/blob/master/src/defaults.rs#L10
Elixir: https://github.com/100phlecs/tailwind_formatter/blob/v0.3.1/priv/classes.txt
Headwind: https://github.com/heybourn/headwind/blob/master/package.json#L57
eslint-plugin-tailwind: https://github.com/Idered/eslint-plugin-tailwind/blob/c13bc81528a3566adaf8cbc418d061f28a498bf0/lib/helpers/default-sort-order.js
Headwind nvim lua plugin: https://github.com/steelsojka/headwind.nvim/tree/b2664dc04b197a9c2c05820c5a28d2089d43b5a0

**Comment by airblade** [comment]
I wrote a standalone script to do this: https://github.com/airblade/tcs.

Unlike Headwind et al., it doesn't use a huge hard-coded list of classes.

**Comment by svoop** [comment]
There's also [Rustywind](https://github.com/avencera/rustywind).
